homeassistant/components/starlink/entity.py

```diff
@@ -0,0 +1,69 @@
+"""Contains base entity classes for Starlink entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DISHY_HARDWARE, DISHY_ID
+from .coordinator import StarlinkUpdateCoordinator
+from .dish_status import DishyStatus
+
+
+@dataclass
+class StarlinkSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[DishyStatus], datetime | StateType]
+
+
+@dataclass
+class StarlinkSensorEntityDescription(
+    SensorEntityDescription, StarlinkSensorEntityDescriptionMixin
+):
+    """Describes a Starlink sensor entity."""
+
+
+class StarlinkSensorEntity(CoordinatorEntity[StarlinkUpdateCoordinator], SensorEntity):
```

Comment:
I think most integrations would have all contents of this file in `sensor.py` although I don't think it's a hard requirement.

Reply:
I plan on having more platforms after this PR, I figure it makes sense to store base entity classes all in one place 😄